File: backend/pkg/domain/service/image.go
package service

import (
	"github.com/gofrs/uuid"
	"server/pkg/domain/model"
)

type ImageService interface {
	StoreImage(imageData string) (model.ImageID, error)
	DeleteImage(imageID model.ImageID) error
}

type imageService struct {
	imageRepo ImageRepository
}

func NewImageService(imageRepo ImageRepository) *imageService {
	return &imageService{
		imageRepo: imageRepo,
	}
}

type ImageRepository interface {
	NextID() uuid.UUID
	Store(image model.Image) error
	Delete(imageID model.ImageID) error
}

func (service *imageService) StoreImage(imageData string) (model.ImageID, error) {
	imageID := model.ImageID(service.imageRepo.NextID())

	image := model.NewImage(
		imageID,
		imageData,
	)

	return imageID, service.imageRepo.Store(image)
}

func (service *imageService) DeleteImage(imageID model.ImageID) error {
	return service.imageRepo.Delete(imageID)
}

File: frontend/src/shared/types/image.ts
export type GetImage = {
  imageId: string;
};

export type Image = {
  imageData: string;
};

export type DeleteImage = {
  imageId: string;
};

export type StoreImageAuthor = {
  authorId: string;
  imageData: string;
};

export type StoreImageBook = {
  bookId: string;
  imageData: string;
};

export type StoreImageUser = {
  imageData: string;
};
